pkg/solana/chainwriter/ata_creation.go

```diff
@@ -129,40 +129,44 @@ func (s *SolanaChainWriterService) handleATACreation(ctx context.Context, create
 		return fmt.Errorf("error enqueuing transaction: %w", err)
 	}
 
-	err = s.waitForTxFinality(ctx, ataUUID)
+	err = s.waitForTxStatus(ctx, ataUUID, types.Unconfirmed)
 	if err != nil {
 		return fmt.Errorf("error waiting for ATA transaction finality: %w", err)
 	}
 	return nil
 }
 
-func (s *SolanaChainWriterService) waitForTxFinality(ctx context.Context, transactionID string) error {
-	waitCtx, cancel := context.WithTimeout(ctx, 5*time.Minute)
+func (s *SolanaChainWriterService) waitForTxStatus(ctx context.Context, transactionID string, desiredStatus types.TransactionStatus) error {
+	waitCtx, cancel := context.WithTimeout(ctx, s.config.TxStatusTimeout.Duration())
 	defer cancel()
 
-	ticker := time.NewTicker(1 * time.Second)
-	defer ticker.Stop()
+	backoff := 1 * time.Second
 
 	for {
 		select {
 		case <-waitCtx.Done():
 			return fmt.Errorf("context ended while waiting for finality of transaction %s", transactionID)
-		case <-ticker.C:
+		case <-time.After(backoff):
 			status, err := s.txm.GetTransactionStatus(waitCtx, transactionID)
 			// Allow poll to continue if status not found for transaction since transactions are not immediately broadcasted
 			// Status check could have started before there is a status to report
 			if err != nil && !strings.Contains(err.Error(), "failed to find transaction") {
 				return fmt.Errorf("error fetching transaction status: %w", err)
 			}
 			switch status {
-			case types.Finalized:
-				s.lggr.Debug("ATA transaction finalized", "transactionID", transactionID)
-				return nil
+			case types.Finalized, types.Unconfirmed:
```

Comment:
Since this method is a little more generalized now, think we should also include `types.Pending` here. Otherwise, calling this method for that status would still wait till Unconfirmed at least

Reply:
good point, added that in as well